### ISSUE TITLE
data analysis with python projects are not i18n friendly

### ISSUE THREAD

**Issue by ilenia-magoni** [freeCodeCamp/freeCodeCamp#47038]
Continuing from https://github.com/freeCodeCamp/boilerplate-medical-data-visualizer/pull/18#discussion_r929168245 by @naomi-lgbt 

All the projects in this certification (somewhere else too maybe?) have many comments inside the boilerplate, resulting that it is missing in i18n.

Project description + boilerplate should be reworked to make so that all the indications are available in the curriculum page so to be translatable.

Possible approach to follow:
don't we have some projects in which the sections in which to write code are numbered and the numbers are referenced in the project description?

**Comment by ieahleen** [comment]
it's the Quality Assurance - Quality Assurance and Testing with Chai section!
> Within tests/1_unit-tests.js under the test labelled #14 in the Strings suite, [...]

For these we could implement something of this kind.

**Comment by Jagrutiti** [comment]
Hey @ieahleen 

I understand that this issue has some translation missing. But I did not understand how to proceed with the solution you have mentioned in the above comment.

**Comment by ieahleen** [comment]
Insted of having comments for each section in which to add code, comments could be numbers instead, and referenced in the project description

**Comment by m0hit-kumar** [comment]
I would like to contribute to this issue, can you assign me the issue

**Comment by ieahleen** [comment]
@m0hit-kumar issues are not assigned generally, feel free to open a PR for all or part of the issue (all the projects may be a lot of work)

**Comment by m0hit-kumar** [comment]
> @m0hit-kumar issues are not assigned generally, feel free to open a PR for all or part of the issue (all the projects may be a lot of work)

can you suggest then what I have to contribute? it will be easy for me to contribute

**Comment by ieahleen** [comment]
The boilerplate code for the Python project is not translatable, all the comments should be moved to the file in the main repo, and the boileplate should be updated so that each section can be referenced.

For example one of the boilerplates has all these comments. https://github.com/freeCodeCamp/boilerplate-medical-data-visualizer/blob/main/medical_data_visualizer.py

What's missing in the project description should be moved to the project description, https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/08-data-analysis-with-python/data-analysis-with-python-projects/medical-data-visualizer.md

And the comments that say "write here" should become something language-agnostic (numbers maybe?) so that the project description can reference those and give the same indications of the current comments in the boilerplate

**Comment by Doraemon23** [comment]
So the solution would be to replace every comment with a number and create a section in the description with information about each comment?

**Comment by ieahleen** [comment]
Yes, the information provided by the comments should instead be provided by the project description

**Comment by Doraemon23** [comment]
Can you check if #47858 is able to fix this problem?

**Comment by ieahleen** [comment]
A possible solution that would not lengthed the project description is to change the comments in the boilerplate to numbers and add the comment number in the Tasks section, as the instructions are already present

**Comment by a2937** [comment]
Hello. I'd like a little more clarification on what's it supposed to look like before I start implementing my take on a solution.

I understand that all the comments in the code will be replaced with numbers. Those numbers then pointing to the instruction file. The instruction file having a new task section which has the comment information written down. 

Did I understand correctly?

**Comment by naomi-lgbt** [comment]
Yes, essentially it would be similar to what we did with the quality assurance lessons.

The comments in the boilerplate should change to numbers, like https://github.com/freeCodeCamp/boilerplate-mochachai/pull/27/commits/9c457443ab770316df5e1b35fb5c7383d57f72a2

Then the instructions in the lesson on the main repo should be updated to reflect the instructions removed from the comments.

**Comment by moT01** [comment]
Well, the sea-level-predictor and the mean-variance-standard-deviation-calculator projects look done. The other three still have quite a few comments in their main files.

**Comment by prachidoddi** [comment]
data analysis with python projects are not i18n friendly #47038
@naomi-lgbt assign me this issue i am ready

**Comment by ilenia-magoni** [comment]
@prachidoddi We typically do not assign issues. Instead, we accept the first pull request that comprehensively solves the issue.

Issues labeled with `help wanted` or `first timers only` are open for contributions.

Please make sure you read [our guidelines for contributing](https://contribute.freecodecamp.org/#/). We prioritize contributors following the instructions in our guide. Join us in [our chat room](https://discord.gg/PRyKn3Vbay) or [the forum](https://forum.freecodecamp.org/c/contributors/3) if you need help contributing - our community will be happy to assist you.
